src/internal/m365/collection/groups/collection.go

```diff
@@ -248,3 +249,211 @@ func (col *prefetchCollection[C, I]) streamItems(ctx context.Context, errs *faul
+		wg            sync.WaitGroup
+		colProgress   chan<- struct{}
+		el            = errs.Local()
+	)
+
+	ctx = clues.Add(ctx, "category", col.Category().String())
+
+	defer func() {
+		close(col.stream)
+		logger.Ctx(ctx).Infow(
+			"finished stream backup collection items",
+			"stats", col.Counter.Values())
+
+		updateStatus(
+			ctx,
+			col.statusUpdater,
+			len(col.added)+len(col.removed),
+			streamedItems,
+			0,
+			col.FullPath().Folder(false),
+			errs.Failure())
+	}()
+
+	if len(col.added)+len(col.removed) > 0 {
+		colProgress = observe.CollectionProgress(
+			ctx,
+			col.Category().HumanString(),
+			col.LocationPath().Elements())
+		defer close(colProgress)
+	}
+
+	semaphoreCh := make(chan struct{}, col.Opts().Parallelism.ItemFetch)
+	defer close(semaphoreCh)
+
+	// delete all removed items
+	for id := range col.removed {
+		semaphoreCh <- struct{}{}
+
+		wg.Add(1)
+
+		go func(id string) {
+			defer wg.Done()
+			defer func() { <-semaphoreCh }()
+
+			col.stream <- data.NewDeletedItem(id)
+
+			atomic.AddInt64(&streamedItems, 1)
+			col.Counter.Inc(count.StreamItemsRemoved)
+
+			if colProgress != nil {
+				colProgress <- struct{}{}
+			}
+		}(id)
+	}
+
+	// add any new items
+	for id, modTime := range col.added {
+		if el.Failure() != nil {
+			break
+		}
+
+		wg.Add(1)
+		semaphoreCh <- struct{}{}
+
+		go func(id string, modTime time.Time) {
+			defer wg.Done()
+			defer func() { <-semaphoreCh }()
+
+			ictx := clues.Add(
+				ctx,
+				"item_id", id,
+				"parent_path", path.LoggableDir(col.LocationPath().String()))
+
+			col.stream <- data.NewLazyItemWithInfo(
+				ictx,
+				&lazyItemGetter[C, I]{
+					modTime:       modTime,
+					getAndAugment: col.getAndAugment,
+					userID:        col.protectedResource,
+					itemID:        id,
+					containerIDs:  col.FullPath().Folders(),
+					contains:      col.contains,
+					parentPath:    col.LocationPath().String(),
+				},
+				id,
+				modTime,
+				col.Counter,
+				el)
+
+			// Need remaining things, copy from prefetch collection
+			// totalBytes - not available since info gets called later now
+			atomic.AddInt64(&streamedItems, 1)
+
+			if colProgress != nil {
+				colProgress <- struct{}{}
+			}
+		}(id, modTime)
+	}
+
+	wg.Wait()
+}
+
+type lazyItemGetter[C graph.GetIDer, I groupsItemer] struct {
+	getAndAugment getItemAndAugmentInfoer[C, I]
+	userID        string
+	itemID        string
+	parentPath    string
+	containerIDs  path.Elements
+	modTime       time.Time
+	contains      container[C]
+}
+
+func (lig *lazyItemGetter[C, I]) GetData(
+	ctx context.Context,
+	errs *fault.Bus,
+) (io.ReadCloser, *details.ItemInfo, bool, error) {
+	writer := kjson.NewJsonSerializationWriter()
+	defer writer.Close()
+
+	item, info, err := lig.getAndAugment.getItem(
+		ctx,
+		lig.userID,
+		lig.containerIDs,
+		lig.itemID)
+	if err != nil {
+		// If an item was deleted then return an empty file so we don't fail
+		// the backup and return a sentinel error when asked for ItemInfo so
```

Comment:
This comment doesn't feel accurate, because we're not returning an error.  Is it outdated?

Reply:
Yep. will fix it.